### PR TITLE
Update examples to use bslib 

### PR DESCRIPTION
**Pull request by skaltman**
This PR updates `runExample()` to use `{bslib}`-powered examples. These new examples are located under `inst/shiny` instead of `inst/examples`. 

To run the old (legacy) examples, set `options(shiny.legacy.examples = TRUE)`

### CODE REVIEW

File: inst/shiny/11_timer/app.R [hunk 1]
```diff
@@ -0,0 +1,22 @@
+library(shiny)
+library(bslib)
+
+# Define UI for displaying current time ----
+ui <- page_fillable(
```
Comment (cpsievert) [inline inst/shiny/11_timer/app.R:5]:
```suggestion
ui <- page_fluid(
```

**Comment by cpsievert** [comment]
Looking good to so far! I just changed file locations around a bit for better backwards-compatibility reasons. 

I also suggested a couple small tweaks, and left them unresolved in case you need to make updates elsewhere

File: inst/shiny/01_hello/app.R [hunk 2]
```diff
@@ -0,0 +1,48 @@
+library(shiny)
+library(bslib)
+
+# Define UI for app that draws a histogram ----
+ui <- page_sidebar(
+
```
Comment (cpsievert) [inline inst/shiny/01_hello/app.R:6]:
```suggestion
```

File: NEWS.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
...
 
 * Both `conditionalPanel()` and `uiOutput()` are now styled with `display: contents` by default in Shiny apps that use Bootstrap 5. This means that the elements they contain are positioned as if they were direct children of the parent container holding the `conditionalPanel()` or `uiOutput()`. This is probably what most users intend when they use these functions, but it may break apps that applied styles directly to the container elements created by these two functions. In that case, you may include CSS rules to set `display: block` for the `.shiny-panel-conditional` or `.shiny-html-output` classes. (#3957, #3960)
+* The examples behind `runExample()` now use `{bslib}` to generate a better looking result. To instead run the "legacy" examples, set `options(shiny.legacy.examples = TRUE)` before calling `runExample()`. (#3963)
```
Comment (cpsievert) [inline NEWS.md:8]:
```suggestion

* The examples behind `runExample()` now use `{bslib}` to generate a better looking result. To instead run the "legacy" examples, set `options(shiny.legacy.examples = TRUE)` before calling `runExample()`. (#3963)
```

File: R/runapp.R [hunk 4]
```diff
@@ -467,7 +467,9 @@ runExample <- function(example=NA,
                        launch.browser = getOption('shiny.launch.browser', interactive()),
                        host=getOption('shiny.host', '127.0.0.1'),
                        display.mode=c("auto", "normal", "showcase")) {
-  examplesDir <- system_file('examples', package='shiny')
+  legacy <- getOption('shiny.legacy.examples', FALSE)
+  examplesDir <- if (isTRUE(legacy)) 'examples' else 'shiny'
+  examplesDir <- system_file(examplesDir, package='shiny')
```
Comment (gadenbuie) [inline R/runapp.R:472]: @karangattu started working on extending `runExample()` to other packages in #3917.

That said, while we're here, I think we should do a very light-weight version of that work and add `package = "shiny"` to `runExample()`'s parameters.

I'll take that in a follow-up PR.

**Review by gadenbuie — APPROVED** [review]
Looks great thanks everyone!

I'd love to see us use consistent and modern code style throughout. I [certainly have opinions](https://github.com/gadenbuie/grkstyle) and am happy to restyle the new examples in a follow up.

In particular, I'd like to follow these rules:

1. All arguments to a function are either on the same line or each argument is on a new line.
2. Closing braces and parens have the same level of indentation as line containing the opening function call.

```r
# Before
      textInput(inputId = "caption",
                label = "Caption:",
                value = "Data Summary"),

      # Input: Selector for choosing dataset ----
      selectInput(inputId = "dataset",
                  label = "Choose a dataset:",
                  choices = c("rock", "pressure", "cars")),

# After
      textInput(
        inputId = "caption",
        label = "Caption:",
        value = "Data Summary"
      ),

      # Input: Selector for choosing dataset ----
      selectInput(
        inputId = "dataset",
        label = "Choose a dataset:",
        choices = c("rock", "pressure", "cars")
      ),
```

Importantly, this code style matches Python's black formatting rules so it's easier for us to be consistent across languages and projects if we follow it.

**Pull request merged**